vllm/model_executor/models/deepseek_v2.py

```diff
@@ -276,9 +290,15 @@ def forward(
         kv_cache: torch.Tensor,
         attn_metadata: AttentionMetadata,
     ) -> torch.Tensor:
+        if is_hpu:
+            # need reshape from tensor(x0, y0) to tensor(x1) for hpu
+            _batch_size = positions.shape[0]
+            positions = positions.reshape(positions.shape[0] * positions.shape[1])
+            hidden_states = hidden_states.reshape(hidden_states.shape[0] * hidden_states.shape[1], hidden_states.shape[2])
         if self.q_lora_rank is not None:
-            q = self.q_a_proj(hidden_states)[0]
-            q = self.q_a_layernorm(q)
+            # w/a of SW-208144
+            q = self.q_a_proj(hidden_states)[0].unsqueeze(0)
```

Comment:
This should be under if_hpu

Reply:
https://github.com/HabanaAI/vllm-fork/pull/516